### ISSUE TITLE
salt-cloud fails to prepend root_dir to log_file

### ISSUE THREAD

**Issue by brianthelion** [saltstack/salt#44366]
### Description of Issue/Question

With non-standard `root_dir`, the `salt-cloud` CLI fails and throws and non-zero exit code if the `log_file` setting isn't specifically set in the `cloud` config. It's not immediately clear from the code where the problem lies.

### Setup

```bash
MINION_NAME='test_minion'
PROFILE_NAME='test_profile'

LOG_LVL=all

ROOT_DIR=$(mktemp -d)
CONFIG_DIR=$ROOT_DIR/etc/salt
mkdir -p $CONFIG_DIR
FILE_ROOT_BASE=$(mktemp -d)

cat <<EOF > $CONFIG_DIR/cloud
root_dir: $ROOT_DIR
#log_file: $ROOT_DIR/var/log/salt/cloud
EOF

salt-cloud --log-file-level=$LOG_LVL --config-dir=$CONFIG_DIR -p $PROFILE_NAME $MINION_NAME
echo "EXIT $?"
```

gives 

```
No permissions to access "/var/log/salt/cloud", are you running as the correct user?
EXIT 1
```

### Steps to Reproduce Issue
Run the script with and without the comment on the `log_file: ...` line.

### Versions Report
```
Salt Version:
           Salt: 2017.7.2
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.7
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.3 (default, Sep 14 2017, 22:58:41)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.3
           RAET: Not Installed
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.2
            ZMQ: 4.1.6
 
System Versions:
           dist: Ubuntu 17.04 zesty
         locale: UTF-8
        machine: x86_64
        release: 4.12.0-041200rc6-generic
         system: Linux
        version: Ubuntu 17.04 zesty

```

**Comment by Ch3LL** [comment]
looks like i'm able to replicate this and we will need to get it fixed. ping @saltstack/team-cloud any ideas on how to fix this?

**Comment by gtmanfred** [comment]
This should be super easy, we just have to add log_file to here.

https://github.com/saltstack/salt/blob/develop/salt/config/__init__.py#L2657

**Comment by brianthelion** [comment]
@gtmanfred I looked at all other calls to the function `prepend_root_dir(...)` and none of them involve `log_file`. It seems that logs are getting taken care of somewhere else in the code base, but I'm not sure exactly where.

TL;DR It's not obvious that that's the right place to patch the code.

**Comment by gtmanfred** [comment]
We do it on other things here.

https://github.com/saltstack/salt/blob/develop/salt/config/__init__.py#L3645 

for the minion config, it gets added to the prepend_root_dirs for the salt minion.

That is the correct place for this to happen.

**Comment by brianthelion** [comment]
Good eye! Previous comment redacted.

**Comment by rallytime** [comment]
@brianthelion This should be fixed with #48182
